src/utils/storage/index.ts

```diff
@@ -10,7 +10,7 @@ export const storage = new ImmortalStorage(stores)
 
 const PREFIX = `v2_${getNetwork()}`
 
-export const loadFromStorage = async (key) => {
+export const loadFromStorage = async <T extends () => unknown>(key: string): Promise<T> => {
```

Comment:
why does it extend a function? should it be `Record<key, unknown>`? We never return a function from there

Reply:
also noticed
```
    if (stringifiedValue === null || stringifiedValue === undefined) {
      return undefined
    }
```
so should be T | undefined then

Reply:
If I don't put it like that I got this error:

![image](https://user-images.githubusercontent.com/21086218/85345476-f4199e80-b4c8-11ea-85b6-5a94c0b61b01.png)


Couldn't find a way to fix it, if you have suggestions feel free to add them

Reply:
@Agupane the message is pretty straightforward, walletSelect has type `string` as an argument and `lastUsedProvider` is of type `() => unknown`

Sorry, I don't want to sound mean but maybe it's worth reading https://basarat.gitbook.io/typescript/?